Sources/Common/Core/DataArray/index.js

```diff
@@ -9,6 +9,31 @@ const TUPLE_HOLDER = [];
 // Global methods
 // ----------------------------------------------------------------------------
 
+// Original source from https://www.npmjs.com/package/compute-range
+// Modified to accept type arrays
+function fastComputeRange(arr) {
+  const len = arr.length;
+  let min;
+  let max;
+  let x;
+  let i;
+
+  if (len === 0) {
+    return null;
+  }
+  min = arr[0];
+  max = min;
+  for (i = 1; i < len; i++) {
```

Comment:
I think you will have to pass `offset` and `numberOfComponents` as arguments and do `i=offset` and `i += numberOfComponents` to properly handle multi-channel data.

Reply:
I sent a commit that _should_ address multi-channel data too, may you take a look please?